source/features/show-local-user-time.tsx

```diff
@@ -0,0 +1,184 @@
+import optionsStorage from '../options-storage';
+import observeEl from '../libs/simplified-element-observer';
+import {clock} from '../libs/icons';
+
+interface Commit {
+	url: string;
+}
+interface Event {
+	type: string;
+	payload: AnyObject;
+	actor: {id: number};
+}
+
+// NOTE: This is basically copied from api.v3() but we can't use it because
+// it always parses the res as json it also can't handle absolute urls yet.
+// TODO: Add this to lib/api somehow
+const settings = optionsStorage.getAll();
+const api3 = location.hostname === 'github.com' ?
+	'https://api.github.com/' :
+	`${location.origin}/api/v3/`;
+const api = async (query: string, options: RequestInit = {}): Promise<Response> => {
+	const {personalToken} = await settings;
+
+	const url = query.startsWith('http') ? query : (api3 + query);
+	const headers = {
+		'User-Agent': 'Refined GitHub',
+		Accept: 'application/vnd.github.v3+json',
+		...options.headers,
+		...(personalToken ? {Authorization: `token ${personalToken}`} : {})
+	};
+
+	return fetch(url, {
+		...options,
+		headers
+	});
+};
+
+async function loadLastCommit(url: string): Promise<Commit|null> {
+	const response = await api(url);
+	if (!response.ok) {
+		return null;
+	}
+
+	// NOTE: We iterate in series here to reduce requests
+	// The first commit should already work in most cases
+	const events = await response.json() as Event[];
+	for (const event of events) {
+		if (event.type !== 'PushEvent') {
+			continue;
+		}
+
+		// NOTE: We want to iterate the commits in reverse to start with the latest commit
+		const commits = event.payload.commits as Commit[];
+		for (let i = commits.length - 1; i >= 0; i--) {
+			const commit = commits[i];
+			// eslint-disable-next-line no-await-in-loop
+			const response = await api(commit.url);
+			if (!response.ok) {
+				// NOTE: Ignore 404 errors and check the next commit
+				if (response.status === 404) {
+					continue;
+				}
+
+				return null;
+			}
+
+			// eslint-disable-next-line no-await-in-loop
+			const {author} = await response.json();
+			// NOTE: Some commits don't have an author when there is no github user for the autor email.
+			if (author && author.id === event.actor.id) {
+				return commit;
+			}
+		}
+	}
+
+	if (!response.headers.has('link')) {
+		return null;
+	}
+
+	const [, next] = (/<([^>]+)>; rel="next"/).exec(response.headers.get('link')!) || [];
+	if (!next) {
+		return null;
+	}
+
+	return loadLastCommit(next);
+}
+
+const getLastCommit = (login: string): Promise<Commit|null> => loadLastCommit(`users/${login}/events`);
+async function getCommitPatch(commit: Commit): Promise<string> {
+	const response = await api(commit.url, {headers: {Accept: 'application/vnd.github.v3.patch'}});
+	return response.ok ? response.text() : '';
+}
+
+// IDEA: We could also return the date from the patch
+// This could help to identify "wrong" offsets e.g. daylight saving
+async function loadTimezoneOffset(login: string): Promise<number> {
```

Comment:
```suggestion
const loadTimezoneOffset = mem(async (login: string): Promise<number> => {
```

`mem` will take care of the caching and you can drop `getTimezoneOffset` altogether

Reply:
Really like that solution :+1: